File: selection.cpp
#include<iostream>
#include<climits>
using namespace std;

int main(){
    cout<<"Enter size of the array :\n";
    int n;
    cin>>n;
    int arr[n];
    cout<<"Enter Array elements :\n";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }

    for(int i=0;i<n-1;i++){
        int min =arr[i];
        int indexOfMin=i;
        for(int j=i+1;j<n;j++){
            if(arr[j]<=min){
            min =arr[j];
            indexOfMin=j;
            }
           
        }
        int temp =arr[i];
        arr[i]=min;
        arr[indexOfMin]=temp;
    }

    for(int z=0;z<n;z++){
        cout<<arr[z]<<" ";
    }

    return 0;
}

File: Selection.java
import java.util.Scanner;

class Selection {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr [] = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        for(int i =0;i<n;i++){
            int min = Integer.MAX_VALUE;
            int indexOfMin = -1;

            for(int j =i;j<n;j++){
                if(arr[j]<min){
                    min = arr[j];
                    indexOfMin = j;
                }
            }
            //swap
            int temp = arr[i];
            arr[i]  = min;
            arr[indexOfMin] = temp;
        }

        for(int i =0;i<n;i++){
            System.out.print(arr[i]+" ");
        }
        sc.close();
    }
}